libecho/include/EchoProtocol.h

```diff
@@ -42,19 +47,32 @@ class EchoProtocol {
 
     /**
      * @brief Closes a connection.
+     * Connection that was once opened should be closed (even when one of read/write
+     * functions throws an exception).
      */
     void close();
 
     /**
-     * Reads at most count bytes from audio input.
-     * Can receive 0 bytes, but if it doesn't receive any signal for timeout seconds,
-     * it throws an exception.
+     * @brief           Reads at most count bytes from audio input and fills buffer with them.
+     * @param buf       pointer to buffer that should be filled
+     * @param count     number of bytes to read
+     * @param timeout   number of seconds; if no signal is detected by then, function assumes
+     *                  that connection is broken;
+     *                  timeout < 0 means that function can wait infinitely
+     *
+     * @return          Returns number of read bytes (can be any from 0 to count),
+     *                  or -1 in case of failure or end of stream.
      */
     ssize_t read(void *buf, size_t count, int timeout);
 
     /**
      * Writes count bytes to audio output.
      * Can throw an exception when sent data isn't being confirmed.
+     *
+     * @param buf     pointer to buffer with data
+     * @param count   number of bytes to send
+     *
+     * @return        Returns number of sent bytes (should be equal to count)
```

Comment:
Może wyjaśnij co zrobić jeśli return_val != count, czy wtedy należy wysyłać jeszcze raz czy co zrobić

Reply:
nie może tak w ogóle być

Reply:
To po co ta wartość zwracana, skoro zawsze jest równa argumentowi?